src/ui/public/visualize/visualization.js

```diff
@@ -38,13 +38,15 @@ uiModules
 
         // Set the passed in uiState to the vis object, so we don't require any
         // users of the <visualization/> directive to manually set the uiState.
-        if (!$scope.uiState) $scope.uiState = new PersistedState({});
+        if (!$scope.uiState) {
+          if ($scope.vis.hasUiState()) {
```

Comment:
what if vis has a uiState before this gets called ?
what if `$scope.uiState` is different from the one set on vis ? which one should prevail ?

Reply:
(currently the one set on visualize/visualization will override the one set on Vis)